packages/shared/src/components/HorizontalScroll/HorizontalScroll.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  RefObject,
+  MouseEventHandler,
+} from 'react';
+import classNames from 'classnames';
+import { useHorizontalScrollHeader } from './useHorizontalScrollHeader';
+
+interface HorizontalScrollProps {
+  title: string | ReactElement;
+  children: ReactNode;
+  onScroll?: (ref: RefObject<HTMLElement>) => void;
+  onClickSeeAll?: MouseEventHandler;
+  className?: string;
+}
+
+export default function HorizontalScroll({
+  title,
+  children,
+  onScroll,
+  onClickSeeAll,
+  className,
+}: HorizontalScrollProps): ReactElement {
+  const { ref, Header } = useHorizontalScrollHeader({
+    title,
+    onScroll,
+    onClickSeeAll,
+  });
+
+  const titleId = `horizontal-scroll-title-${Math.random()}`;
+
+  return (
+    <>
```

Comment:
I think we should make this an actual wrapper of the header and content to make it more predictable.
```suggestion
    <div className="flex flex-col">
```

Reply:
Yeah makes sense, thank u! will add